influxdb3/tests/server/auth.rs

```diff
@@ -1,78 +1,27 @@
-use parking_lot::Mutex;
+use arrow_flight::error::FlightError;
+use arrow_util::assert_batches_sorted_eq;
+use influxdb3_client::Precision;
 use reqwest::StatusCode;
-use std::env;
-use std::mem;
-use std::panic;
-use std::process::Child;
-use std::process::Command;
-use std::process::Stdio;
-
-struct DropCommand {
-    cmd: Option<Child>,
-}
-
-impl DropCommand {
-    const fn new(cmd: Child) -> Self {
-        Self { cmd: Some(cmd) }
-    }
 
-    fn kill(&mut self) {
-        let mut cmd = self.cmd.take().unwrap();
-        cmd.kill().unwrap();
-        mem::drop(cmd);
-    }
-}
-
-static COMMAND: Mutex<Option<DropCommand>> = parking_lot::const_mutex(None);
+use crate::{collect_stream, TestServer};
 
 #[tokio::test]
 async fn auth() {
     const HASHED_TOKEN: &str = "5315f0c4714537843face80cca8c18e27ce88e31e9be7a5232dc4dc8444f27c0227a9bd64831d3ab58f652bd0262dd8558dd08870ac9e5c650972ce9e4259439";
     const TOKEN: &str = "apiv3_mp75KQAhbqv0GeQXk8MPuZ3ztaLEaR5JzS8iifk1FwuroSVyXXyrJK1c4gEr1kHkmbgzDV-j3MvQpaIMVJBAiA";
-    // The binary is made before testing so we have access to it
-    let bin_path = {
-        let mut bin_path = env::current_exe().unwrap();
-        bin_path.pop();
-        bin_path.pop();
-        bin_path.join("influxdb3")
-    };
-    let server = DropCommand::new(
-        Command::new(bin_path)
-            .args([
-                "serve",
-                "--object-store",
-                "memory",
-                "--bearer-token",
-                HASHED_TOKEN,
-            ])
-            .stdout(Stdio::null())
-            .stderr(Stdio::null())
-            .spawn()
-            .expect("Was able to spawn a server"),
-    );
 
-    *COMMAND.lock() = Some(server);
-
-    let current_hook = panic::take_hook();
-    panic::set_hook(Box::new(move |info| {
-        COMMAND.lock().take().unwrap().kill();
-        current_hook(info);
-    }));
+    let server = TestServer::configure()
+        .auth_token(HASHED_TOKEN, TOKEN)
+        .spawn()
+        .await;
 
     let client = reqwest::Client::new();
-
-    // Wait for the server to come up
-    while client
-        .get("http://127.0.0.1:8181/health")
-        .bearer_auth(TOKEN)
-        .send()
-        .await
-        .is_err()
-    {}
+    let base = server.client_addr();
 
     assert_eq!(
         client
-            .post("http://127.0.0.1:8181/api/v3/write_lp?db=foo")
+            .post(format!("{base}/api/v3/write_lp"))
+            .query(&[("db", "foo")])
```

Comment:
It's a bit confusing to be doing a write but the function is called query. I guess this is part of the `reqwest` client but it's confusing in a db test suite. Wouldn't we be able to use `influxdb3_client` here? Or was your thinking that it's kind of outside the scope of these changes and we can do a follow up PR? I think issue https://github.com/influxdata/influxdb/issues/24694 would cover this.

Reply:
Or even `write_lp_to_db` which we use in the other test

Reply:
I mainly switched to using the [`query` method](https://docs.rs/reqwest/latest/reqwest/struct.RequestBuilder.html#method.query) on `reqwest`'s `RequestBuilder` for the sake of the below calls that were passing `select * from cpu` to make them a bit easier to read. It is perhaps less necessary here, but I changed it throughout for consistency.

I think its okay to keep using `reqwest` here as it affords us to more fine grained control over how we interact with the API, and allows us to closely test the behaviour of the API, i.e., status codes returned. We could use the `influxdb3_client`, which would have the added benefit of also testing the client, but it would change the nature of the assertions being made, and I wanted to keep this test focused on the HTTP API itself.

Reply:
@hiltontj okay yeah that makes a lot of sense to me, especially with badly written headers being the purpose of the test. Thanks for the clarification!